framework/base/Model.php

```diff
@@ -216,7 +216,7 @@ public function scenarios()
         }
 
         foreach ($scenarios as $scenario => $attributes) {
-            if (empty($attributes) && $scenario !== self::SCENARIO_DEFAULT) {
+            if ($scenario !== self::SCENARIO_DEFAULT && empty($attributes)) {
```

Comment:
Possibility that scenario is default is high, so this code will work faster in 99% cases.